Automattic/jetpack
# Related Posts: Add support for author name in block editor

Fixes #30823.

## Proposed changes:


1. Adds a toggle to the block editor to "Display author".
2. Generates author string during `render_block()` call.
3. Renders author in resulting block in both front/backend.

The resulting UI should look like this:

<img width="1023" alt="SCR-20230520-pvtc" src="https://github.com/Automattic/jetpack/assets/40267301/12f0cedb-1df2-4ebd-9797-05b00850fc0e">

### Other information:


## Jetpack product discussion



pejTkB-yT-p2

## Does this pull request change what data or activity we track or use?



No

## Testing instructions:






1. Spin up a test site with this branch.
3. Make sure Related Posts are enabled via Jetpack → Settings → Traffic → Related Posts.
4. Create a new post and add a Related Posts block.
5. Confirm the "Display author" toggle is present and disabled by default.
6. Confirm enabling the toggle updates the block in the editor.
7. Confirm the author is shown on the front-facing page once published.

**Note:** You may need to publish before the placeholders are replaced with actual post content.
